src/shared/Microsoft.Git.CredentialManager/Interop/Posix/PosixEnvironment.cs

```diff
@@ -29,6 +31,35 @@ protected override string[] SplitPathVariable(string value)
             return value.Split(':');
         }
 
+        public override string LocateExecutable(string program)
```

Comment:
This implementation and the Windows one below are nearly duplicates with the only difference the newlines to split on and the location of the locator program.

We could move this to the abstract class and add abstract methods for defining the locator program and the array of chars to split lines on.

Reply:
We could, but that then assumes that we're shelling out to an executable to tell us where applications can be found, rather than say walking the `%PATH%`/`$PATH` ourselves, or calling some OS API or looking in the registry.

In 'old' GCM there's a considerable (too much IMO) amount of logic for locating files and installations of things like Git, but there was a need to do this. Things like portable Git or Git in WSL may become more important.

Reply:
That's fair, if this was dupped a 3rd time then I'd say we have real duplication on hand.